hadoop-tools/hadoop-aliyun/src/test/java/org/apache/hadoop/fs/aliyun/oss/TestAliyunOSSRedirect.java

```diff
@@ -0,0 +1,264 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.fs.aliyun.oss;
+
+// Standard Java imports
+import java.io.IOException;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+
+// SLF4J imports
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+// Apache imports
+import org.apache.commons.lang3.StringUtils;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FSDataInputStream;
+import com.aliyun.oss.common.auth.CredentialsProvider;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.IOUtils;
+import org.apache.hadoop.util.VersionInfo;
+
+// Aliyun OSS imports
+import com.aliyun.oss.ClientConfiguration;
+import com.aliyun.oss.OSSClient;
+import com.aliyun.oss.OSSException;
+import com.aliyun.oss.common.comm.Protocol;
+import com.aliyun.oss.model.RoutingRule;
+import com.aliyun.oss.model.SetBucketWebsiteRequest;
+import com.aliyun.oss.ClientException;
+
+// JUnit imports
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.internal.AssumptionViolatedException;
+import org.junit.rules.Timeout;
+import static org.junit.Assert.assertThrows;
+import static org.junit.Assert.assertTrue;
+import static org.apache.hadoop.fs.aliyun.oss.Constants.*;
+
+/**
+ * This test checks the redirection behavior of Aliyun OSS.
+ * When the redirection feature is available, it will access the
+ * redirected target file
+ */
+public class TestAliyunOSSRedirect {
```

Comment:
I'm going to to suggest subclassing AbstractHadoopTestBase

This
* adds the test timeout, and supports a system property to extend it
* sets the thread name to the test method, different names for setup and teardown. *this is really useful*
Once you do that you can cut line 81; nothing else should change

Reply:
Thank you, steveloughran, for the review. I will carefully revise these comments and submit the improved code as soon as possible.